src/csvcubed/utils/pandas.py

```diff
@@ -15,35 +15,41 @@
 
 from csvcubed.models.cube.validationerrors import DuplicateColumnTitleError
 
-specified_na_values = {
+
+_logger = logging.getLogger(__name__)
+
+# Values used in place of NA in dataframe reads
+SPECIFIED_NA_VALUES = {
     "",
 }
 
-logger = logging.getLogger(__name__)
 
-def read_csv(csv: Path, **kwargs) -> Tuple[pd.DataFrame, List[ValidationError]]:
+def read_csv(csv_path: Path, **kwargs) -> Tuple[pd.DataFrame, List[ValidationError]]:
     """
     :returns: a tuple of
         pd.DataFrame without the default na values being changes into NaN
         list of ValidationExceptions
     """
-    df = pd.read_csv(
-        csv, keep_default_na=False, na_values=specified_na_values, **kwargs
-    )
+    # Set default but allow interventions for advanced users
+    if "keep_default_na" not in kwargs:
```

Comment:
I take your meaning but want to quickly double check a point in case it got lost in the code.

so kwargs is _any keywords you could pass to pandas_, (there's tons), I'm trying to keep the ability to finesse the csv read if/when we need to, there's an awful lot of power we can leverage with that.

I could do something like:

```
get_pandas_datatypes(
    csv_path: Path, config: Optional[dict] = None, keep_default_na: Bool = False, na_values: dict =specified_na_values, **kwargs 
) -> Dict[str, str]:
```

which'd work, but it sits slightly oddly in my head as then kwargs is all pandas kwargs _except_ for the two we're explicitly hinting.

and yeah, something about what those kwargs are should be in the docstring regardless, the fact I'm double checking kinda implies I missed a trick there.

no actual horse in this race btw, I can see a case for declaring the most pertinent kwargs so happy to take the steer, just seemed prudent to check we're on the same page first.

Reply:
Ok, I can see what you're trying to do there now. 

I guess I'm in two minds about it as well. On the one hand it may be helpful for the caller to be able to finesse how pandas reads CSV in, but at the same time I'm aware that we may not want to use pandas in the longer term, particularly when we start addressing how we're going to cope with larger inputs. So part of me wants to limit how much further we allow pandas to become enmeshed into our code base (even though I will admit it's pretty enmeshed already!).

Could you stick with passing named parameters around for now and if we want to pass any other params to pandas, then we can explicitly add them as needed?

Reply:
What I'm thinking is we can use something like [blaze](https://blaze.readthedocs.io/en/latest/quickstart.html#stored-data) which has a pandas dataframe-like interface but can be backed by other datastores like sqlite. That way when we read the data in, we wouldn't use pandas `read_csv` and instead would load the CSV directly into a sqlite database and then ask blaze to expose it as a dataframe for us. I suspect that would almost give us a drop-in replacement.